### PR TITLE
feat: change instrumentation related k8s objects

### PR DESCRIPTION
**Pull request by mat-rumian**
##### Description

* rename `otelagent` to `otelcol-instrumentation` switch from `daemonset` to `statefulset`
* rename `otelcol` to `traces-sampler`
* rename `otel-gateway` to `traces-gateway`
* moves processors like `resource`, `source`, `resourcedetection` from `traces-sampler` to `otelcol-instrumentation`
* moves `metrics` exporter from `traces-sampler` to `otelcol-instrumentation`
* cleanup services (deprecated `....otelcol` and `....otelagent` point to `otelcol-instrumentation` statefulset)
* cleanup old `traces` values used in `fluentd`
* enable port `8888` for `otelcol-logs` and `otelcol-metrics` by default
* updates otel-distro version to `0.57.2-sumo-1`

---

##### Checklist

- [x] Changelog updated
- [x] Helm tests updated
- [x] Integration tests updated
- [x] Cluster deployment tests (instrumentation flow)
- [x] Parameters list updated

###### Testing performed

- [x] Deployment on test cluster with enabled instrumentation
- [x] Deployment on test cluster with opentelemetry-operator enabled

### CODE REVIEW

**Review by swiatekm-sumo — COMMENTED** [review]
The changes themselves look fine to me. The two questions I have are:
* are you going to add a config migration for the changes to values.yaml
* and/or are you going to describe this in the migration guide

Seems like quite the change at first glance.

**Review by sumo-drosiek — APPROVED** [review]

**Review by swiatekm-sumo — APPROVED** [review]
As far as I can tell, this looks good, but I'd prefer if someone else reviewed it as well.

File: deploy/helm/sumologic/templates/instrumentation/otelcol-instrumentation/statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+{{- if $otelcolInstrumentation.statefulset.extraVolumeMounts }}
+{{ toYaml $otelcolInstrumentation.statefulset.extraVolumeMounts | indent 8 }}
+{{- end }}
+        env:
+{{- $ctx := .Values -}}
+{{- include "kubernetes.sources.env" (dict "Context" $ctx "Type" "metrics" "Name" "default" ) | nindent 8 -}}
+{{- if $otelcolInstrumentation.statefulset.extraEnvVars }}
+{{- toYaml $otelcolInstrumentation.statefulset.extraEnvVars | nindent 8 }}
+{{- end }}
+{{- end }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/instrumentation/otelcol-instrumentation/statefulset.yaml:165]:
```suggestion
{{- end }}

```

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/instrumentation/otelcol-instrumentation/statefulset.yaml:165]: :joy:

File: deploy/helm/sumologic/templates/instrumentation/traces-sampler/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "sumologic.metadata.name.tracessampler.configmap" . }}
+  labels:
+    app: {{ template "sumologic.labels.app.tracessampler.configmap" . }}
+    {{- include "sumologic.labels.common" . | nindent 4 }}
+data:
+  {{- (tpl (.Files.Glob "conf/instrumentation/traces.sampler.conf.yaml").AsConfig .) | nindent 2 }}
+{{- end }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/instrumentation/traces-sampler/configmap.yaml:12]:
```suggestion
{{- end }}

```

File: deploy/helm/sumologic/templates/logs/fluentd/configmap.yaml [hunk 3]
```diff
@@ -9,9 +9,6 @@ metadata:
 data:
   fluent.conf: |-
 {{- .Values.fluentd.logs.rawConfig | nindent 4 }}
-{{- if .Values.sumologic.traces.enabled }}
-    @include traces.conf
-{{- end }}
 {{- (tpl (.Files.Glob "conf/fluentd/*.conf").AsConfig .) | nindent 2 }}
 {{- (tpl (.Files.Glob "conf/logs/fluentd/*.conf").AsConfig .) | nindent 2 }}
 {{- end }}
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/templates/logs/fluentd/configmap.yaml:14]:
```suggestion
{{- end }}

```

File: tests/helm/metadata_logs_fluentd/static/default_v2.output.yaml [hunk 4]
```diff
@@ -146,6 +146,10 @@ data:
   
     
   
+    
+  
+    
+
```
Comment (sumo-drosiek) [inline tests/helm/metadata_logs_fluentd/static/default_v2.output.yaml:152]: I don't like it

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**